File: library/core/src/main/java/com/lacker/micros/domain/exception/WrapperAppException.java
package com.lacker.micros.domain.exception;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * 外套异常
 */
public class WrapperAppException extends ApplicationException {

    private final Map<String, Object> data = new LinkedHashMap<>();

    public WrapperAppException(Throwable throwable) {
        super(throwable.getMessage(), throwable);
    }

    @Override
    public StackTraceElement[] getStackTrace() {
        return super.getCause().getStackTrace();
    }

    public Map<String, Object> getData() {
        return data;
    }

    public void addData(String key, String value) {
        data.put(key, value);
    }
}

File: library/core/src/main/kotlin/com/mlacker/micros/domain/exception/WrapperAppException.kt
package com.mlacker.micros.domain.exception

import java.util.*

/**
 * 外套异常
 */
class WrapperAppException(throwable: Throwable) : ApplicationException(throwable.message, throwable) {
    var data: MutableMap<String, Any> = LinkedHashMap()

    override fun getStackTrace(): Array<StackTraceElement> = cause!!.stackTrace
}